script/DeployMigrationStakeManager.s.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.19 <=0.9.0;
+
+import { BaseScript } from "./Base.s.sol";
+import { DeploymentConfig } from "./DeploymentConfig.s.sol";
+import { StakeManager } from "../contracts/StakeManager.sol";
+import { VaultFactory } from "../contracts/VaultFactory.sol";
+
+contract DeployMigrationStakeManager is BaseScript {
+    error DeployMigrationStakeManager_InvalidStakeTokenAddress();
+
+    address public prevStakeManager;
+
+    address public stakeToken;
+
+    constructor(address _prevStakeManager, address _stakeToken) {
+        prevStakeManager = _prevStakeManager;
+        stakeToken = _stakeToken;
+    }
+
+    function run() public returns (StakeManager) {
+        prevStakeManager = vm.envOr({ name: "PREV_STAKE_MANAGER", defaultValue: prevStakeManager });
+        stakeToken = vm.envOr({ name: "STAKE_TOKEN_ADDRESS", defaultValue: stakeToken });
+
+        if (stakeToken == address(0)) {
+            revert DeployMigrationStakeManager_InvalidStakeTokenAddress();
+        }
```

Comment:
Theoretically, we can extend this and say, if `prevStakeManager != address(0)`, then we query the `stakeToken()` address from the old manager.